common/app/views/support/cleaner/AmpAdCleaner.scala

```diff
@@ -106,18 +106,17 @@ case class AmpAdCleaner(edition: Edition, uri: String, article: Article) extends
         debug = environment.isNonProd
       )}></amp-ad>
 
-    val ampAdString = {
+    def withDataBlockOnConsent(ampAd: Elem): String =
       // data-block-on-consent should not have ANY value
       // according to https://www.ampproject.org/docs/reference/components/amp-consent
       // This is not compatible with proper XML, hence the stringware hack
+      ( ampAd % Attribute(null, "data-block-on-consent", "PLEASEREMOVEME", Null) ).toString().replaceFirst("=\"PLEASEREMOVEME\"", "")
+
+    val ampAdString = {
+      val usAdRegionSlot = withDataBlockOnConsent(ampAd(UsAdRegion))
+      val auAdRegionSlot = withDataBlockOnConsent(ampAd(AuAdRegion))
+      val rowAdRegionSlot = withDataBlockOnConsent(ampAd(RowAdRegion))
```

Comment:
Is there any straightforward way to test these, with and without consent?

Reply:
Good point. I need to check that.

Reply:
I've tried setting the consent component to show up only when my geo is 'us', so I don't get the consent prompt on my dev box (as it's in gb). The ads show straight away (so the 'data-block-on-consent' is neutered).